File: service/api/src/main/java/org/delivery/api/domain/token/controller/model/TokenResponse.java
package org.delivery.api.domain.token.controller.model;

import java.time.LocalDateTime;

import org.delivery.api.domain.token.model.TokenDto;

import lombok.Builder;

@Builder
public record TokenResponse(
	String accessToken,
	LocalDateTime accessTokenExpiredAt,
	String refreshToken,
	LocalDateTime refreshTokenExpiredAt
) {
	public static TokenResponse of(final TokenDto accessToken, final TokenDto refreshToken) {
		return TokenResponse.builder()
			.accessToken(accessToken.token())
			.accessTokenExpiredAt(accessToken.expiredAt())
			.refreshToken(refreshToken.token())
			.refreshTokenExpiredAt(refreshToken.expiredAt())
			.build();
	}
}

File: service/account/src/main/kotlin/org/delivery/account/domain/token/controller/model/TokenResponse.kt
package org.delivery.account.domain.token.controller

import org.delivery.account.domain.token.model.TokenDto
import java.time.LocalDateTime

data class TokenResponse(
    val accessToken: String,
    val accessTokenExpiredAt: LocalDateTime,
    val refreshToken: String,
    val refreshTokenExpiredAt: LocalDateTime
) {

    companion object {
        @JvmStatic
        fun of(accessToken: TokenDto, refreshToken: TokenDto): TokenResponse {
            return TokenResponse(
                accessToken = accessToken.token!!,
                accessTokenExpiredAt = accessToken.expiredAt!!,
                refreshToken = refreshToken.token!!,
                refreshTokenExpiredAt = refreshToken.expiredAt!!
            );
        }
    }

}
